src/Snake.java

```diff
@@ -0,0 +1,272 @@
+import java.util.ArrayList;
+import java.util.List;
+import edu.macalester.graphics.GraphicsGroup;
+import edu.macalester.graphics.GraphicsObject;
+import edu.macalester.graphics.Point;
+
+
+public class Snake extends GraphicsGroup {
+
+    private int i = 0;
+    private double speed = 2;
+
+    private double dx = 0;
+    private double dy = 0;
+
+    private static double width = 15;
+
+    private boolean movingLeft;
+    private boolean movingRight;
+    private boolean movingUp;
+    private boolean movingDown;
+
+    private GraphicsObject snakeHeadLeft = SnakeHead.createSnakeHead(270);
+    private GraphicsObject snakeHeadRight = SnakeHead.createSnakeHead(90);
+    private GraphicsObject snakeHeadUp = SnakeHead.createSnakeHead(0);
+    private GraphicsObject snakeHeadDown = SnakeHead.createSnakeHead(180);
+    private GraphicsObject snakeHead;
+
+    /**
+     * Sets the current snake head (snakeHead) to snakeHeadUp, initializes all of the movement booleans to false,
+     * and sets the starting position of the snake.
+     */
+    public Snake() {
+        snakeHead = snakeHeadUp;
+        this.add(snakeHead);
+        movingLeft = false;
+        movingRight = false;
+        movingUp = false;
+        movingDown = false;
+        this.startPosition();
+    }
+
+    /**
+     * Moves the snake's head left, turns on the movement boolean for the snake moving left, and calls the method to 
+     * turn the snake's head appropriately (turnSnakeHead();)
```

Comment:
I appreciate your thorough Javadoc! When you write it, try to talk a bit less about how a method/class is implemented, a little more about what it does. For example, I’d write this as: “Causes any future snake movement to be in a leftward direction, and points the snake head graphic to the left.”